lib/lift.ts

```diff
@@ -20,9 +21,12 @@ export function lift<TArgs extends unknown[], TReturn>(
 ): LiftedFn<TArgs, TReturn> {
   return (...args: TArgs) => {
     return ({
-      [Symbol.iterator]() {
-        let value = fn(...args);
-        return { next: () => ({ done: true, value }) };
+      *[Symbol.iterator]() {
+        return yield () => {
+          return shift(function* (k) {
+            k({ ok: true, value: fn(...args) });
```

Comment:
should this be `k.tail` since you don't care about the return value of `k`?

Reply:
Nice catch!